WireAPI/Sources/WireAPI/Models/UserClient/MLSPublicKeys.swift

```diff
@@ -48,8 +48,20 @@ public struct MLSPublicKeys: Equatable, Codable, Sendable {
         case ed448
         case p256 = "ecdsa_secp256r1_sha256"
         case p384 = "ecdsa_secp384r1_sha384"
-        case p512 = "ecdsa_secp512r1_sha512"
+        case p512 = "ecdsa_secp521r1_sha512"
 
     }
 
 }
+
+public extension MLSPublicKeys {
+
+    func hasValidKeys() -> Bool {
+        [ed25519, ed448, p256, p384, p512].contains { key in
+            guard let key else { return false }
+
+            return !key.isEmpty
+        }
```

Comment:
Non blocking suggestion:

```suggestion
    func hasValidKey() -> Bool {
        [ed25519, ed448, p256, p384, p512].contains { key in
            guard let key else { return false }
            return !key.isEmpty
        }
```